zerver/models/realms.py

```diff
@@ -625,6 +625,9 @@ class Realm(models.Model):  # type: ignore[django-manager-missing] # django-stub
     # Whether clients should display "(guest)" after names of guest users.
     enable_guest_user_indicator = models.BooleanField(default=True)
 
+    # Whether to notify client when a DM has a guest recipient.
+    enable_guest_user_notification = models.BooleanField(default=True)
```

Comment:
`enable_guest_user_dm_warning` might be better; started https://chat.zulip.org/#narrow/stream/378-api-design/topic/guest.20user.20warning.20.2330222/near/1810306 to discuss/decide on a convention.

Reply:
Done! But this has caused the PR checks to fail due to migration issue.